lib/ua_update_messaging.py

```diff
@@ -0,0 +1,317 @@
+            apps_pkg_file,
+            apps_no_pkg_file,
+            motd_apps_pkg_file,
+            motd_apps_no_pkg_file,
+        )
+    _write_esm_service_msg_templates(
+        cfg,
+        infra_inst,
+        expiry_status,
+        remaining_days,
+        infra_pkg_file,
+        infra_no_pkg_file,
+        motd_infra_pkg_file,
+        motd_infra_no_pkg_file,
+    )
+
+
+def write_esm_announcement_message(cfg: config.UAConfig, series: str) -> None:
+    """Write human-readable messages if ESM is offered on this LTS release.
+
+    Do not write ESM announcements on trusty, esm-apps is enable or beta.
+
+    :param cfg: UAConfig instance for this environment.
+    :param series: string of Ubuntu release series: 'xenial'.
+    """
+    apps_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME["esm-apps"]
+    apps_inst = apps_cls(cfg)
+    enabled_status = ApplicationStatus.ENABLED
+    apps_not_enabled = apps_inst.application_status()[0] != enabled_status
+    config_allow_beta = util.is_config_value_true(
+        config=cfg.cfg, path_to_value="features.allow_beta"
+    )
+    apps_not_beta = bool(config_allow_beta or not apps_cls.is_beta)
+
+    msg_dir = os.path.join(cfg.data_dir, "messages")
+    esm_news_file = os.path.join(msg_dir, ExternalMessage.ESM_ANNOUNCE.value)
+    if all([series != "trusty", apps_not_beta, apps_not_enabled]):
+        util.write_file(esm_news_file, MESSAGE_ANNOUNCE_ESM)
+    else:
+        util.remove_file(esm_news_file)
+
+
+def update_apt_and_motd_messages(cfg: config.UAConfig) -> None:
+    """Emit templates and human-readable status messages in msg_dir.
+
+    These structured messages will be sourced by both /etc/update.motd.d
+    and APT UA-configured hooks. APT hook content will orginate from
+    apt-hook/hook.cc
+
+    Call esm-apt-hook process-templates to render final humnan-readable
+    messages.
+
+    :param cfg: UAConfig instance for this environment.
+    """
+    setup_logging(logging.INFO, logging.DEBUG)
+    logging.debug("Updating UA messages for APT and MOTD.")
+    msg_dir = os.path.join(cfg.data_dir, "messages")
+    if not os.path.exists(msg_dir):
+        os.makedirs(msg_dir)
+
+    series = util.get_platform_info()["series"]
+    config_allow_beta = util.is_config_value_true(
+        config=cfg.cfg, path_to_value="features.allow_beta"
+    )
+    is_active_esm = bool(config_allow_beta or util.is_active_esm(series))
+
+    if not all([util.is_lts(series), is_active_esm]):
+        # ESM is only on LTS releases. Remove all messages and templates.
+        for msg_enum in ExternalMessage:
+            msg_path = os.path.join(msg_dir, msg_enum.value)
+            util.remove_file(msg_path)
+            if msg_path.endswith(".tmpl"):
+                util.remove_file(msg_path.replace(".tmpl", ""))
+        return
+
+    # Announce ESM availabilty on active ESM LTS releases
+    write_esm_announcement_message(cfg, series)
+    write_apt_pre_invoke_templates(cfg)
```

Comment:
Do we still want to run `apt-esm-hook process-templates` after this? or just let it happen when the user runs `apt update` next?

Reply:
@orndorffgrant I think it's probably a good idea to run process-tempates just to sync with current state and avoid strange "racy" behavior if apt pkg counts change. Good suggestion.